compiler/modules/types/list.bal

```diff
@@ -334,7 +341,16 @@ function fixedArrayFill(FixedLengthArray arr, int newLen, SemType filler) {
     if newLen <= initial.length() {
         return;
     }
-    if arr.fixedLength == 0 || initial[initial.length() - 1] != filler {
+    int initLen = initial.length();
```

Comment:
This was buggy before, right? Do we have a test case?

Reply:
I observe this in the debugger, let me add a test

Reply:
Test cases in `proj9-t.bal` are failing without this fix. Specifically `T4[1]`.